server/server/src/static_files.rs

```diff
@@ -148,17 +148,26 @@ async fn upload_sync_file(
         )
     })?;
 
-    let static_file = upload_sync_file_inner(&service_provider, &settings, path.into_inner(), file)
-        .await
-        .map_err(|error| {
-            log::error!("Error while uploading file: {}", format_error(&error));
-            InternalError::new(
-                "Error uploading file, please check server logs",
-                StatusCode::INTERNAL_SERVER_ERROR,
-            )
-        })?;
-
-    Ok(HttpResponse::Ok().json(static_file.id))
+    let path_inner = path.into_inner();
+
+    let mut static_file_ids: Vec<String> = [].to_vec();
+
+    for f in file {
+        let static_file =
+            upload_sync_file_inner(&service_provider, &settings, path_inner.clone(), f)
```

Comment:
Was trying to pass a reference to `path` rather than cloning here, but the called function takes a tuple of values here, so wasn't sure what the correct syntax to pass reference to object that is being destructured.

Reply:
FWIW, I tried this and the syntax is like this:   

 (table_name, record_id): &(String, String),

Then you pass it here as:

```suggestion
            upload_sync_file_inner(&service_provider, &settings, &path_inner, f)
```

However, I don't think it actually make is more efficient or simplier as you then need to call the `to_string()` or `to_owned()` methods on those variables in the `upload_sync_file_inner` function, so I think what you have is totally fine.